serializers: minor fixes to datacite json and add tests for xml

ppanero opened the pull request:
- closes https://github.com/inveniosoftware/invenio-rdm-records/issues/703
- closes https://github.com/inveniosoftware/invenio-rdm-records/issues/727

Differences I can observe in comparison with the full example of [DataCite 4.3 XML](https://github.com/inveniosoftware/datacite/blob/master/tests/data/datacite-v4.3-full-example.xml), which are not yet implemented are:
- All languages should use the 2 letter encoding, instead of 3. This should be fixed for the JSON schema too.
- We do not have alternate identifiers, we put them in `identifiers`. This would differ from the JSON schema. We should:
    - pids --> identifiers
    - identifiers --> alternate identifiers

ppanero commented:
> All languages should use the 2 letter encoding, instead of 3. This should be fixed for the JSON schema too.

BCP 47 includes ISO 639-3 so we are good.

ppanero commented:
> We do not have alternate identifiers, we put them in identifiers. This would differ from the JSON schema. We should:
pids --> identifiers
identifiers --> alternate identifiers

Not needed, is taken care by datacite, [ref](https://github.com/inveniosoftware/datacite/blob/master/datacite/schema43.py#L58)

lnielsen commented:
Other issues I've seen:

1) Dates should also serialize the embargo date.
![Screenshot 2021-07-07 at 15 02 19](https://user-images.githubusercontent.com/1698163/124763514-5575a980-df34-11eb-9dca-7ffef01db3fe.png)

ppanero commented:
Rebased on latest master (There were conflicts) and added the content of https://github.com/inveniosoftware/invenio-rdm-records/pull/731 to close https://github.com/inveniosoftware/invenio-rdm-records/issues/727

lnielsen commented on tests/resources/serializers/test_datacite_serializer.py:
```diff
@@ -257,3 +261,88 @@ def test_datacite43_serializer(running_app, full_record):
+        "  </formats>",
+        "  <version>v1.0</version>",
+        "  <rightsList>",
+        "    <rights rightsURI=\"https://creativecommons.org/licenses/by/4.0/\" rightsIdentifierScheme=\"spdx\" rightsIdentifier=\"cc-by-4.0\">Creative Commons Attribution 4.0 International</rights>",  # noqa
```
The tests are failing here however. I saw another PR that made a change the record ``rights`` property, thus that's likely the cause. However, the string above is correct, and we should serialize as above, so we'll need to check how to fix it.

ppanero replied:
It was modified to make tests passed. @jrcastro2 in your PR they should pass with 
```
 "  <rightsList>",
"    <rights rightsURI=\"https://creativecommons.org/licenses/by/4.0/\" rightsIdentifierScheme=\"spdx\" rightsIdentifier=\"cc-by-4.0\">Creative Commons Attribution 4.0 International</rights>",  # noqa
"  </rightsList>",
```
You might need to add something to the conftests values maybe? Worth to add a custom one too?

lnielsen reviewed (approved):
Besides the "rigths" field issue, all is good.
